pkg/ruler/manager.go

```diff
@@ -0,0 +1,231 @@
+package ruler
+
+import (
+	"context"
+	"net/http"
+	"sync"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	ot "github.com/opentracing/opentracing-go"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/notifier"
+	promRules "github.com/prometheus/prometheus/rules"
+	"github.com/weaveworks/common/user"
+	"golang.org/x/net/context/ctxhttp"
+
+	store "github.com/cortexproject/cortex/pkg/ruler/rules"
+	"github.com/cortexproject/cortex/pkg/util"
+)
+
+type MultiTenantManager interface {
```

Comment:
This should go next to Ruler, which uses the interface. Please add comments to this interface and its methods.